examples/go_over_there.py

```diff
@@ -0,0 +1,72 @@
+""" pg.examples.go_over_there
+This simple tech demo is showcasing the use of Vector2.move_towards()
+using multiple circles to represent Vectors. Each circle will have a
+random position and speed once the demo starts.
+
+Mouse Controls:
+* Use the mouse to click on a new target position
+
+Keyboard Controls:
+* Press R to restart the demo
+"""
+import pygame as pg
+import random
+
+MIN_SPEED = 0.25
+MAX_SPEED = 5
+MAX_BALLS = 1600
+SCREEN_SIZE = pg.Vector2(1000, 600)
+CIRCLE_RADIUS = 5
+
+pg.init()
+screen = pg.display.set_mode(SCREEN_SIZE)
+clock = pg.time.Clock()
+
+target_position = None
+balls = []
+
+class Ball:
+    def __init__(self, position, speed):
+        self.position = position
+        self.speed = speed
+
+def reset():
+    global balls
+    global target_position
+    
+    target_position = None
+    balls = []
+    for x in range(MAX_BALLS):
+        pos = pg.Vector2(random.randint(0, SCREEN_SIZE.x), random.randint(0, SCREEN_SIZE.y))
+        speed = random.uniform(MIN_SPEED, MAX_SPEED)
+
+        b = Ball(pos, speed)
+        balls.append(b)
+
+reset()
+delta_time = 0
+running = True
+while running:
+    for event in pg.event.get():
+        if event.type == pg.QUIT:
+            running = False
+
+        if event.type == pg.MOUSEBUTTONUP:
+            target_position = pg.mouse.get_pos()
+
+        if event.type == pg.KEYUP:
+            if event.key == pg.K_r:
```

Comment:
Suggestion: I like to provide a keyboard exit shortcut too like pressing escape (e.g. as on windows 'alt+F4' to close the program). AFAIK `pg.QUIT` is only triggered when pressing the 'x' button on the window frame so its mouse only.

Resolve at will.